spec/models/info_request_spec.rb

```diff
@@ -50,13 +50,254 @@
+                  :handle_rejected_responses => 'holding_pen' }
+        info_request = FactoryGirl.create(:info_request_with_incoming, attrs)
+        email, raw_email = email_and_raw_email(:from => 'bob@example.com')
+        info_request.receive(email, raw_email)
+        expect(info_request.reload.incoming_messages.size).to eq(2)
+      end
+
+      it 'from authority_only rejects if there is no from address' do
+        attrs = { :allow_new_responses_from => 'authority_only',
+                  :handle_rejected_responses => 'holding_pen' }
+        info_request = FactoryGirl.create(:info_request, attrs)
+        email, raw_email = email_and_raw_email(:from => '')
+        info_request.receive(email, raw_email)
+        expect(info_request.reload.incoming_messages.size).to eq(0)
+        holding_pen = InfoRequest.holding_pen_request 
+        expect(holding_pen.incoming_messages.size).to eq(1)
+        msg = 'Only the authority can reply to this request, but there is ' \
+              'no "From" address to check against'
+        expect(holding_pen.info_request_events.last.params[:rejected_reason]).
+          to eq(msg)
+      end
+
+      it 'from authority_only rejects if the mail is not from the authority' do
+        attrs = { :allow_new_responses_from => 'authority_only',
+                  :handle_rejected_responses => 'holding_pen' }
+        info_request = FactoryGirl.create(:info_request, attrs)
+        email, raw_email = email_and_raw_email(:from => 'spam@example.net')
+        info_request.receive(email, raw_email)
+        expect(info_request.reload.incoming_messages.size).to eq(0)
+        holding_pen = InfoRequest.holding_pen_request 
+        expect(holding_pen.incoming_messages.size).to eq(1)
+        msg = "Only the authority can reply to this request, and I don't " \
+              "recognise the address this reply was sent from"
+        expect(holding_pen.info_request_events.last.params[:rejected_reason]).
+          to eq(msg)
+      end
+
+      it 'raises an error if there is an unknown allow_new_responses_from' do
+        info_request = FactoryGirl.create(:info_request)
+        info_request.allow_new_responses_from = 'unknown_value'
+        email, raw_email = email_and_raw_email
+        err = InfoRequest::ResponseGatekeeper::UnknownResponseGatekeeperError
+        expect { info_request.receive(email, raw_email) }.
+          to raise_error(err)
+      end
+
+      it 'can override the stop new responses status of a request' do
+        attrs = { :allow_new_responses_from => 'nobody',
+                  :handle_rejected_responses => 'holding_pen' }
+        info_request = FactoryGirl.create(:info_request, attrs)
+        email, raw_email = email_and_raw_email
+        info_request.receive(email, raw_email, true)
+        expect(info_request.incoming_messages.size).to eq(1)
+      end
+
+      it 'does not check spam when overriding the stop new responses status of a request' do
```

Comment:
Line is too long. [92/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.